006-connected-udp-sockets.md

```diff
@@ -0,0 +1,67 @@
+| Title  | Connected UDP sockets |
+|--------|-----------------------|
+| Author | @santigimeno          |
+| Status | DRAFT                 |
+| Date   | 2017-03-25 21:41:00   |
+
+
+## Overview
+
+Calling `connect()` on a UDP socket associates a remote address and port to the
+socket, so every message sent by this socket is automatically sent to that
+destination.
+
+## API
+
+The following functions are defined:
+
+```c
+int uv_udp_connect(uv_udp_t* handle, const struct sockaddr* addr);
+
+int uv_udp_disconnect(uv_udp_t* handle);
```

Comment:
Quoting again *Stevens UNIX Network Programming:*

> The most portable solution is to zero out an address structure, set the family to AF_UNSPEC as mentioned above, and pass it to connect.

I have implemented `uv_disconnect()` like this and seems to work fine on: Linux, OS X, FreeBSD and Windows.

The only issue I've found is that the bind state of the socket after being disconnected differs between platforms. For example on Linux the socket address and port are set to 0 whereas on FreeBSD the address is set to 0 but the port is set to the port it was initially bound.